streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -135,11 +147,18 @@ public class RecordCollectorTest {
     private StreamsProducer streamsProducer;
     private ProcessorTopology topology;
     private final InternalProcessorContext<Void, Void> context = new InternalMockProcessorContext<>();
-
     private RecordCollectorImpl collector;
+    final Sensor mockSensor = Mockito.mock(Sensor.class);
 
     @Before
     public void setup() {
+
+        when(mockStreamsMetrics.taskLevelSensor(Mockito.anyString(), Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Sensor.RecordingLevel.class), Mockito.any(Sensor[].class))).thenReturn(mockSensor);
```

Comment:
nit: 
Just for better readability
```suggestion
        when(
            mockStreamsMetrics.taskLevelSensor(
                Mockito.anyString(), 
                Mockito.anyString(), 
                Mockito.anyString(),
                Mockito.any(Sensor.RecordingLevel.class), 
                Mockito.any(Sensor[].class)
        )).thenReturn(mockSensor);
```